### PR TITLE
Setup backend

### PR DESCRIPTION
**Pull request by FabianAlleblas**
Time for a tiny backend, I just wrote some functions in the PlantRepository.php and the PlantController.php.
The rest is generated by symfony and doctrine, oh yeah and a validation in the Plant.php.
I know that I have to write my own serializer at some point, but for now it's working :).

Changes in the frontend:
getData.js
postData.ja
Error handling in the usePlantList hook.

### CODE REVIEW

File: backend/src/Repository/PlantRepository.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Plant;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+
+/**
```
Comment (macfrei) [inline backend/src/Repository/PlantRepository.php:9]: I don't understand anything about PHP, however, it looks like you have a lot of comments in your files. I'd delete them before merging.

**Review by macfrei — COMMENTED** [review]
The front-end looks good to me! The only thing I noticed are a lot of comments in your PHP files. Keep up the great work 🚀

File: backend/api-request.http [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+DELETE {{baseUrl}}/plant/12
+
+###
+
+PUT {{baseUrl}}/plant/5
+
+  {
+    "plantname": "Hello and thx for reviewing this important part of my capstone project :D",
+    "plantspecies": "Let me tell you a story, how I lost 1 hour of my time, just because I had a , on the wrong place of my test-json down below."
```
Comment (leonp5) [inline backend/api-request.http:23]: Business as usual ;-)
Googling a day and then deciding to try a complete different approach...

File: backend/src/Repository/PlantRepository.php (hunk 1 above)
Comment (leonp5) [reply backend/src/Repository/PlantRepository.php:9]: As far as i understand it, the comments are helpful for the IDE and the code quality tools

File: frontend/src/data/plants.json [hunk 3]
```diff
@@ -1,32 +1,32 @@
 [
   {
     "id": 1,
-    "name": "Carl",
-    "species": "Monstera deliciosa"
+    "plantname": "Carl",
```
Comment (alex-jonas) [inline frontend/src/data/plants.json:4]: Is it just for testing or why do you save this information in the frontend?

File: frontend/src/services/getData.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const baseUrl = 'http://urbanplants.local/plant'
+
+export default async function getData() {
+  try {
+    const response = await fetch(baseUrl)
+    const data = response.json()
+    return data
+  } catch (error) {
+    return { error: 'The server is down! :(' }
```
Comment (leonp5) [inline frontend/src/services/getData.js:9]: Why you don't write the errors in the console?
`catch (error) {
console.error(error)
}`
Might be helpful if it doesnt work...

File: frontend/src/hooks/usePlantList.js [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
 
 export default function usePlantList() {
-  const [plantList, setPlantList] = useState(loadFromLocal('PlantList') ?? [])
+  const [plantList, setPlantList] = useState([])
 
   useEffect(() => {
-    saveToLocal('PlantList', plantList)
-  }, [plantList])
+    getData().then((data) =>
+      data.error ? alert(data.error) : setPlantList(data)
```
Comment (alex-jonas) [inline frontend/src/hooks/usePlantList.js:10]: What speaks against `.catch` for error-handling?

File: backend/api-request.http [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# vscode extension "Humao Rest Client"
```
Comment (alex-jonas) [inline backend/api-request.http]: You could add this to gitignore

File: frontend/src/Components/Forms/AddPlantForm.js [hunk 7]
```diff
@@ -11,9 +11,8 @@ export default function AddPlantForm({ savePlantData }) {
 
   const history = useHistory()
   const initialInput = {
-    id: '',
-    name: '',
-    species: '',
+    plantname: '',
+    plantspecies: '',
```
Comment (alex-jonas) [inline frontend/src/Components/Forms/AddPlantForm.js:15]: Wouldn't it have been simpler to name the object more meaningful? (initialPlant or so)

**Pull request merged**